pypeit/core/wavecal/wvutils.py

```diff
@@ -199,81 +204,81 @@ def get_wave_grid(waves, masks=None, wave_method='linear', iref=0, wave_grid_min
+            nexp = waves.shape[1]
+            newloglam = loglam[:, iref]  # Deals with mask
+            # Loop
+            for j in range(nexp):
+                if j == iref:
+                    continue
+                #
+                iloglam = loglam[:, j]
+                dloglam_0 = (newloglam[1]-newloglam[0])
+                dloglam_n =  (newloglam[-1] - newloglam[-2]) # Assumes sorted
+                if (newloglam[0] - iloglam[0]) > dloglam_0:
+                    kmin = np.argmin(np.abs(iloglam - newloglam[0] - dloglam_0))
+                    newloglam = np.concatenate([iloglam[:kmin], newloglam])
+                #
+                if (iloglam[-1] - newloglam[-1]) > dloglam_n:
+                    kmin = np.argmin(np.abs(iloglam - newloglam[-1] - dloglam_n))
+                    newloglam = np.concatenate([newloglam, iloglam[kmin:]])
+            # Finish
+            wave_grid = np.power(10.0,newloglam)
+
+        elif 'iref' in wave_method:
+            wave_tmp = waves[:, iref]
+            wave_grid = wave_tmp[ wave_tmp > 1.0]
 
-    elif 'linear' in wave_method: # Constant Angstrom
-        if dwave is not None:
-            dwave_pix = dwave
         else:
-            dwave_pix = dwave_data
-        # Generate wavelength array
-        wave_grid = wavegrid(wave_grid_min, wave_grid_max, dwave_pix, spec_samp_fact=spec_samp_fact)
-        wave_grid_mid = wave_grid + dwave_pix*spec_samp_fact/2.0
-        dsamp = dwave_pix
-
-    elif 'concatenate' in wave_method:  # Concatenate
-        # Setup
-        loglam = np.log10(waves) # This deals with padding (0's) just fine, i.e. they get masked..
-        nexp = waves.shape[1]
-        newloglam = loglam[:, iref]  # Deals with mask
-        # Loop
-        for j in range(nexp):
-            if j == iref:
-                continue
-            #
-            iloglam = loglam[:, j]
-            dloglam_0 = (newloglam[1]-newloglam[0])
-            dloglam_n =  (newloglam[-1] - newloglam[-2]) # Assumes sorted
-            if (newloglam[0] - iloglam[0]) > dloglam_0:
-                kmin = np.argmin(np.abs(iloglam - newloglam[0] - dloglam_0))
-                newloglam = np.concatenate([iloglam[:kmin], newloglam])
-            #
-            if (iloglam[-1] - newloglam[-1]) > dloglam_n:
-                kmin = np.argmin(np.abs(iloglam - newloglam[-1] - dloglam_n))
-                newloglam = np.concatenate([newloglam, iloglam[kmin:]])
-        # Finish
-        wave_grid = np.power(10.0,newloglam)
-
-    elif 'iref' in wave_method:
-        wave_tmp = waves[:, iref]
-        wave_grid = wave_tmp[ wave_tmp > 1.0]
+            msgs.error("Bad method for wavelength grid: {:s}".format(wave_method))
 
-    else:
-        msgs.error("Bad method for wavelength grid: {:s}".format(wave_method))
-
-    if ('iref' in wave_method) | ('concatenate' in wave_method):
+    if ('iref' in wave_method) | ('concatenate' in wave_method) | ('user_input' in wave_method):
```

Comment:
I prefer

`if wave_method in ['iref', 'concatenate', 'user_input']:`

easier to parse

Reply:
Changed.